src/background.js

```diff
@@ -84,40 +92,34 @@ class ImageClassifier {
     }
   }
 
-  /**
-   * Triggers the model to make a prediction on the image referenced by url.
-   * After a successful prediction a IMAGE_CLICK_PROCESSED message when
-   * complete, for the content.js script to hear and update the DOM with the
-   * results of the prediction.
-   *
-   * @param {string} url url of image to analyze.
-   * @param {number} tabId which tab the request comes from.
-   */
-  async analyzeImage(url, tabId) {
-    if (!tabId) {
-      console.error('No tab.  No prediction.');
-      return;
-    }
-    if (!this.model) {
-      console.log('Waiting for model to load...');
-      setTimeout(() => {this.analyzeImage(url)}, FIVE_SECONDS_IN_MS);
-      return;
-    }
-    let message;
-    this.loadImage(url).then(
+  async analyzeImage(url) {
+    chrome.tabs.query({ active: true }, (tabs) => {
+      const tabId = tabs[0].id
+      if (!tabId) {
+        console.error('No tab.  No prediction.');
+        return;
+      }
+      if (!this.model) {
+        console.log('Waiting for model to load...');
+        setTimeout(() => { this.analyzeImage(url) }, FIVE_SECONDS_IN_MS);
```

Comment:
wtf is this can u pls make into a promise

Reply:
dude this was here before lmao

Reply:
Ye LOOL it's fine